sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -848,4 +851,24 @@ object DDLUtils {
       }
     }
   }
+
+  private[sql] def checkFieldNames(table: CatalogTable): Unit = {
+    table.provider.get.toLowerCase(Locale.ROOT) match {
```

Comment:
How about first identify the format and then do the check?

Reply:
Do you mean `table.storage.serde` by `the format`?

Reply:
y.